otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -895,13 +895,10 @@ static value caml_runtime_events_user_resolve_cached(
 
     // copy values from the cache to the new array
     for (uintnat i = 0; i < len; i++) {
-      Field(cache_resized, i) = Field(cache, i);
+      caml_initialize(&Field(cache_resized, i), Field(cache, i));
```

Comment:
isn't `caml_initialize` only needed if allocated via `caml_alloc_shr`?  `cache_resized` is allocated via `caml_alloc`

Reply:
Yes and no: `caml_alloc` calls `caml_alloc_shr` when the requested size is too big for allocating on the minor heap.

Reply:
What does "Yes and no" mean here? It's either "yes" or "no", and "no" according to our FFI documentation.

`caml_alloc` initialises all the fields to something that isn't garbage (Val_unit in the case of a larger allocation) after its call to `caml_alloc_shr`.  So you shouldn't need `caml_initialize` here.

Reply:
I meant yes to the first part understood as "caml_initialize (instead of direct assignment) is only needed for allocations on the major heap", and no to what is implied by the second part (caml_alloc can also allocate on the major heap, and so cannot use a direct assignment).

Semantics aside, here the code would naively use `caml_modify` to copy the fields, but using `caml_initialize` is a valid optimization over `caml_modify` in this case.